jetstream_pt/engine.py

```diff
@@ -570,7 +584,19 @@ def generate(
     )
 
     next_token = self._sampling(logits, self.env.batch_size)
-    lens = decode_state.lens + 1
+    if self.env.ring_buffer:
+      input_pos = decode_state.input_pos + 1
+      lens = decode_state.lens + 1
+    else:
+      input_pos = jnp.where(
+          decode_state.input_pos == 0,
+          0,
+          decode_state.input_pos + 1 % self.env.cache_len,
```

Comment:
In non ring buffer case, can input_pos be larger than cache len?

Reply:
If no, I feel we don't need do % since it never reach the cache len.

Reply:
We don't have control for this. Generate() will keep running if no new prefill results are inserted.

Reply:
Thanks for sharing the details!